### DISCUSSION TITLE
Image Registry credentials required ?

### DISCUSSION THREAD

**Discussion by adeniyistephen** [aquasecurity/trivy#1078] [Q&A] [answered]
Hello Trivy team, I'm Stephen a mentee of LFX(Linux foundation) summer 21', I'm working on a project for my program which is to develop an Image Vulnerability scanner and output the result in Kubernetes Policy Report CRD. here is the issue: https://github.com/kubernetes-sigs/wg-policy-prototypes/issues/54, also the discussion is on the Kubernetes wg-policy channel https://kubernetes.slack.com/archives/C906A4GAF. 
I have a quick question concerning image registry credentials - why are Image Registry credentials required? Does Trivy actually download the image or does it use the one available in the cluster? (I had to scan the image in my cluster when I pushed the image to docker hub first). 
Thank you, please feel free to join the slack channel.

**Comment by knqyf263** [comment, marked as answer]
First of all, if an image is in a local Docker (or Podman) daemon, Trivy simply uses it. If it doesn't exist locally, Trivy needs to pull the image from a registry. Kubernetes is most likely not using Docker, so Trivy can't scan local images.

FYI: [Starboard](https://github.com/aquasecurity/starboard) is able to levarage imagePullSecrets for pulling an image.

**Reply by adeniyistephen** [reply]
@knqyf263 Thank you for the clarifications, but I have tried to scan an image that has not yet been pushed to docker hub, it's still on my local docker desktop, when I scanned it with trivy I got an error, not until I pushed it to docker hub and it worked then. why?

**Reply by knqyf263** [reply]
Could you give me what commands did you run and an error message?

**Reply by adeniyistephen** [reply]
yes I can, I'm currently running trivy on a docker container tho. so: `docker run ghcr.io/aquasecurity/trivy:0.18.3 image shipa:1.0` but now that I have pushed it to docker hub I can run it `docker run ghcr.io/aquasecurity/trivy:0.18.3 image adeniyistephen/shipa:1.0` so as other images

**Reply by adeniyistephen** [reply]
@knqyf263 are you there?

**Reply by knqyf263** [reply]
You have to mount docker socket if the image is in your host machine.
https://aquasecurity.github.io/trivy/v0.18.3/installation/#docker

**Reply by adeniyistephen** [reply]
Thank you for this, I'm using docker image so I have been trying to find a way to get my cache dir :) .
